js/private/test/node-patches/opendir.mjs

```diff
@@ -69,6 +77,23 @@ describe('testing opendir', async () => {
                     new URL(`file://${path.join(fixturesDir, 'a')}`)
                 )
                 assert.equal(entry1.name, (await dir.read()).name)
+                assert.equal(entry2.name, (await dir.read()).name)
+
+                // Assert reading via opendirSync() produces the same result
+                dir = fs.opendirSync(path.join(fixturesDir, 'a'))
+                const entry1Sync = dir.readSync()
+                const entry2Sync = dir.readSync()
+                assert.equal(entry1.name, entry1Sync.name)
+                assert.equal(
+                    entry1.isSymbolicLink(),
```

Comment:
One of these `isSymbolicLink` calls is what failed before the "fix" commit which overrides `opendirSync` to be like the non-Sync